File: src/domains/puzzle/PuzzleProblem.java
package domains.puzzle;

import domains.dummy.*;
import framework.problem.Problem;



public class PuzzleProblem extends Problem {
    
        public PuzzleProblem() {
            super();
            super.setName("8 Tile Puzzle");
            super.setIntroduction(INTRO);
            super.setMover(new PuzzleMover());
            puzzle[0][0] = 2;
            puzzle[0][1] = 8;
            puzzle[0][2] = 3;
            puzzle[1][0] = 1;
            puzzle[1][1] = 6;
            puzzle[1][2] = 4;
            puzzle[2][0] = 7;
            puzzle[2][1] = 0;
            puzzle[2][2] = 5;
            super.setInitialState(new PuzzleState(puzzle));
            super.setCurrentState(super.getInitialState());
            puzzleFinal[0][0] = 1;
            puzzleFinal[0][1] = 2;
            puzzleFinal[0][2] = 3;
            puzzleFinal[1][0] = 8;
            puzzleFinal[1][1] = 0;
            puzzleFinal[1][2] = 4;
            puzzleFinal[2][0] = 7;
            puzzleFinal[2][1] = 6;
            puzzleFinal[2][2] = 5;
            super.setFinalState(new PuzzleState(puzzleFinal));
        }

        int [][] puzzle = new int [3][3];
        int [][] puzzleFinal = new int [3][3];
        
        private static final String INTRO = 
                "There is a board upon which numbered tiles are placed in a semi"
                + "random order. A legal move is moving a tile into the blank space"
                + "which is adjacent to it. When the puzzle is finished it will look"
                + "like the final board placed below.\n" 
                +  "+---+---+---+\n" +
                  "| 1 | 2 | 3 |\n" +
                  "+---+---+---+\n" +
                  "| 8 |   | 4 |\n" +
                  "+---+---+---+\n" +
                  "| 7 | 6 | 5 |\n" +
                  "+---+---+---+";

    }

File: js/puzzle/PuzzleProblem.js
function PuzzleProblem() {
  this.name = "8-Puzzle Problem";
  this.introduction =  "You are given a board in which numbered tiles can slide around." +
            "There is one blank space that holds no tile." +
            "A legal move consists of sliding a tile into the blank space if the tile is adjacent to it." +
            "The goal is to move tiles around until the board looks like the final state below.\n" +
                "+---+---+---+\n" +
                "| 1 | 2 | 3 |\n" +
                "+---+---+---+\n" +
                "| 8 |   | 4 |\n" +
                "+---+---+---+\n" +
                "| 7 | 6 | 5 |\n" +
                "+---+---+---+";
        
   this.initialState = new PuzzleState([[2,8,3],[1,6,4],[7,0,5]]);
   this.currentState = this.initialState;
   this.finalState = new PuzzleState([[1,2,3],[8,0,4],[7,6,5]]);
   this.moveNames = [SLIDE_1, SLIDE_2, SLIDE_3, SLIDE_4, SLIDE_5, SLIDE_6, SLIDE_7, SLIDE_8];
   this.mover = new PuzzleMover();

    this.success = function () {
	return this.currentState.equals(this.finalState);
    };
}
